bindings/rust/wasmedge-sys/src/config.rs

```diff
@@ -1,79 +1,129 @@
+//! Defines WasmEdge Config struct.
+
 use crate::{
     types::{CompilerOptimizationLevel, CompilerOutputFormat},
     wasmedge, Error, WasmEdgeResult,
 };
 
+/// Struct of WasmEdge Config.
+///
+/// [`Config`] manages the configuration options, which are used in WasmEdge [`Vm`], [`Loader`], [`Validator`],
+/// [`Executor`], and [`Compiler`].
+///
+/// The configuration options are categorized into the following four groups:
+///
+/// - WebAssembly Proposals
+///
+///     This group of options are used to turn on/off the WebAssembly proposals. They are effective to any WasmEdge
+///     context created with [`Config`].
+///     
+///     - `ImportExportMutGlobals` supports mutable imported and exported globals.
+///
+///       Also see [Import/Export Mutable Globals Proposal](https://github.com/WebAssembly/mutable-global/blob/master/proposals/mutable-global/Overview.md#importexport-mutable-globals).
+///
+///     - `NonTrapFloatToIntConversions` supports the non-trapping float-to-int conversion.
+///
+///       Also see [Non-trapping Float-to-int Conversions Proposal](https://github.com/WebAssembly/spec/blob/main/proposals/nontrapping-float-to-int-conversion/Overview.md).
+///
+///     - `SignExtensionOperators` supports new integer instructions for sign-extending 8-bit, 16-bit, and 32-bit values.
+///     
+///       Also see [Sign-extension Operators Proposal](https://github.com/WebAssembly/spec/blob/main/proposals/sign-extension-ops/Overview.md).
+///
+///     - `MultiValue` supports functions and instructions with multiple return values, and blocks with inputs.
+///     
+///       Also see [Multi-value Extension](https://github.com/WebAssembly/spec/blob/main/proposals/multi-value/Overview.md).
+///
+///     - `BulkMemoryOperations` supports bulk memory operations.
+///
+///       Also see [Bulk Memory Operations Proposal](https://github.com/WebAssembly/spec/blob/main/proposals/bulk-memory-operations/Overview.md#motivation-for-bulk-memory-operations).
+///
+///     - `ReferenceTypes` supports reference types.
+///
+///       Also see [Reference Types Proposal](https://github.com/WebAssembly/spec/blob/main/proposals/reference-types/Overview.md).
+///
+///     - `SIMD` supports 128-bit packed SIMD extension to WebAssembly.
+///
+///       Also see [SIMD Proposal](https://github.com/WebAssembly/spec/blob/main/proposals/simd/SIMD.md).
+///  
+///     - `TailCall` supports tail call optimization.
+///
+///       Also see [Tail Call Proposal](https://github.com/WebAssembly/tail-call/blob/master/proposals/tail-call/Overview.md).
+///
+///       Also see [](https://github.com/WebAssembly/tail-call/blob/master/proposals/tail-call/Overview.md).
+///
+///     - `Annotations` supports annotations in WASM text format.
+///
+///       Also see [Annotations Proposal](https://github.com/WebAssembly/annotations/blob/master/proposals/annotations/Overview.md).
+///
+///     - `Memory64` support linear memory of sizes larger than 232 bits.
```

Comment:
Why `232`?

Reply:
@hydai Accepted. It's a typo and should be power(2, 32). Thanks!

Reply:
Will refactor the statement. Thanks!